device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -116,51 +117,30 @@ public AmqpsIotHubConnection(DeviceClientConfig config)
             throw new IllegalArgumentException("hubName cannot be null or empty.");
         }
 
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_15_002: [The constructor shall save the configuration into private member variables.]
         this.deviceClientConfig = config;
 
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_12_017: [The constructor shall set the AMQP socket port using the configuration.]
-        this.useWebSockets = this.deviceClientConfig.isUseWebsocket();
-        if (useWebSockets)
-        {
-            this.hostName = String.format("%s:%d", this.chooseHostname(), AMQP_WEB_SOCKET_PORT);
-        }
-        else
-        {
-            this.hostName = String.format("%s:%d", this.chooseHostname(), AMQP_PORT);
-        }
+        this.hostName = this.chooseHostname();
 
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_15_004: [The constructor shall initialize a new Handshaker
-        // (Proton) object to handle communication handshake.]
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_15_005: [The constructor shall initialize a new FlowController
-        // (Proton) object to handle communication flow.]
         add(new Handshaker());
         add(new FlowController());
 
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_15_006: [The constructor shall set its state to DISCONNECTED.]
         this.state = IotHubConnectionStatus.DISCONNECTED;
 
-        this.amqpsSessionManager = new AmqpsSessionManager(this.deviceClientConfig, this);
-
-        log.trace("AmqpsIotHubConnection object is created successfully and will use port {}", useWebSockets ? AMQP_WEB_SOCKET_PORT : AMQP_PORT);
-    }
-
-    /**
-     * Creates a new DeviceOperation using the given configuration..
-     *
-     * @param deviceClientConfig the device configuration to add.
-     * @throws TransportException if adding the device fails
-     */
-    public void addDeviceOperationSession(DeviceClientConfig deviceClientConfig) throws TransportException
-    {
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_12_018: [The function shall do nothing if the deviceClientConfig parameter is null.]
-        if (deviceClientConfig != null)
+        DeviceClientConfig.AuthType authType = config.getAuthenticationType();
+        if (authType.equals(DeviceClientConfig.AuthType.SAS_TOKEN))
         {
-            // Codes_SRS_AMQPSIOTHUBCONNECTION_12_019: [The function shall call AmqpsSessionManager.addDeviceOperationSession with the given deviceClientConfig.]
-            this.amqpsSessionManager.addDeviceOperationSession(deviceClientConfig);
-
-            this.log.trace("Added device to session list");
+            this.amqpsAuthenticationLinkHandler = new AmqpsAuthenticationLinkHandlerCBS();
+        }
+        else if (authType == DeviceClientConfig.AuthType.X509_CERTIFICATE)
```

Comment:
Q - who sets `DeviceClientConfig.AuthType`? Is it the sdk or the user?

Reply:
The SDK sets it, but only after the user has provided us their certificates or connection string so that we know what type they are using.